core/prelude/types/uint.carbon

```diff
@@ -30,4 +32,160 @@ impl forall [N:! IntLiteral()] UInt(N) as As(IntLiteral()) {
   fn Convert[self: Self]() -> IntLiteral() = "int.convert_checked";
 }
 
-// TODO: Operations.
+// Comparisons.
+
+impl forall [N:! IntLiteral()] UInt(N) as Eq {
+  fn Equal[self: Self](other: Self) -> bool = "int.eq";
+  fn NotEqual[self: Self](other: Self) -> bool = "int.neq";
+}
+
+impl forall [N:! IntLiteral()] UInt(N) as Ordered {
+  // TODO: fn Compare
+  fn Less[self: Self](other: Self) -> bool = "int.less";
+  fn LessOrEquivalent[self: Self](other: Self) -> bool = "int.less_eq";
+  fn Greater[self: Self](other: Self) -> bool = "int.greater";
+  fn GreaterOrEquivalent[self: Self](other: Self) -> bool = "int.greater_eq";
+}
+
+// Arithmetic.
+
+impl forall [N:! IntLiteral()] UInt(N) as Add {
+  fn Op[self: Self](other: Self) -> Self = "int.sadd";
```

Comment:
Should this have been `int.uadd` and similar for the rest below?

Not exactly sure how to write a test for this, what happens if you add past i32::MAX (or however we'd spell that)?

Reply:
A test in `lower/` catches it, and confirms these should be changed: https://github.com/carbon-language/carbon-lang/pull/4740